ravdess/main_msaf.py

```diff
@@ -114,69 +105,71 @@ def get_X(device, sample):
             training_loader = data.DataLoader(training_set, **params)
-            val_set = RAVDESSDataset(val_fold, modality=modalities, transform=val_transform)
-            val_loader = data.DataLoader(val_set, **params)
-
-            print("Fold " + str(i + 1))
             print("Train fold: ")
             print([os.path.basename(act) for act in train_fold])
-            print("val fold: ")
-            print([os.path.basename(act) for act in val_fold])
 
             # record training process
             current_time = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
             train_log_dir = os.path.join(args.checkpointdir, 'logs/fold{}_{}'.format(i + 1, current_time))
             writer = SummaryWriter(log_dir=train_log_dir)
 
-            # define model
-            model_param = {}
-            if "video" in modalities:
-                video_model = resnet50(
-                    num_classes=8,
-                    shortcut_type='B',
-                    cardinality=32,
-                    sample_size=224,
-                    sample_duration=30)
-                video_model_path = os.path.join(args.checkpointdir, "resnext50/fold_{}_resnext50_best.pth".format(i+1))
-                video_model_checkpoint = torch.load(video_model_path) if use_cuda else \
-                    torch.load(video_model_path, map_location=torch.device('cpu'))
-                video_model.load_state_dict(video_model_checkpoint)
-                model_param.update(
-                    {"video": {
-                        "model": video_model,
-                        "id": modalities.index("video")
-                    }})
-
-            if "audio" in modalities:
-                audio_model = MFCCNet()
-                audio_model_path = os.path.join(args.checkpointdir, "mfccNet/fold_{}_mfccNet_best.pth".format(i + 1))
-                audio_model_checkpoint = torch.load(audio_model_path) if use_cuda else \
-                    torch.load(audio_model_path, map_location=torch.device('cpu'))
-                audio_model.load_state_dict(audio_model_checkpoint)
-                model_param.update(
-                    {"audio": {
-                        "model": audio_model,
-                        "id": modalities.index("audio")
-                    }})
-
-            multimodal_model = MSAFNet(model_param)
-            multimodal_model.to(device)
-
+        val_fold = all_folder[i * s: i * s + s]
+        val_set = RAVDESSDataset(val_fold, transform=val_transform)
+        val_loader = data.DataLoader(val_set, **params)
+        print("val fold: ")
+        print([os.path.basename(act) for act in val_fold])
+
+        # define model
+        # video model
+        model_param = {}
+        video_model = resnet50(
+            num_classes=8,
+            shortcut_type='B',
+            cardinality=32,
+            sample_size=224,
+            sample_duration=30)
+        audio_model = MFCCNet()
+        if args.train:
```

Comment:
Maybe we should have another parameter to determine if we should load weights or not for training e.g args.no_load_pretrained which is default False

parser.add_argument('--no_load_pretrained', action='store_false', default=True, help='skip loading weights before training')